crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -1619,6 +1620,41 @@ TEST_P(HKDF_ServiceIndicatorTest, HKDFTest) {
   EXPECT_EQ(approved, test.expect_approved);
 }
 
+// TODO(CryptoAlg-1281): Do we need to test HKDF_Expand separately from this?
+class EVP_HKDF_ServiceIndicatorTest : public TestWithNoErrors<HKDFTestVector> {};
+
+INSTANTIATE_TEST_SUITE_P(All, EVP_HKDF_ServiceIndicatorTest,
+                         testing::ValuesIn(kHKDFTestVectors));
+
+TEST_P(EVP_HKDF_ServiceIndicatorTest, HKDFTest) {
+  const HKDFTestVector &test = GetParam();
+
+  FIPSStatus approved = AWSLC_NOT_APPROVED;
+
+  uint8_t output[sizeof(kHKDF_okm_tc2_sha256)];   // largest test vector output size
+  EVP_PKEY_CTX *pctx;
+  size_t outlen = test.output_len;
+
+  pctx = EVP_PKEY_CTX_new_id(EVP_PKEY_HKDF, NULL);
+  EXPECT_NE(pctx, nullptr);
+  EXPECT_EQ(EVP_PKEY_derive_init(pctx), 1);
+  EXPECT_EQ(EVP_PKEY_CTX_set_hkdf_md(pctx, test.func()), 1);
+  EXPECT_EQ(EVP_PKEY_CTX_set1_hkdf_key(pctx, test.ikm, test.ikm_size), 1);
+  EXPECT_EQ(EVP_PKEY_CTX_set1_hkdf_salt(pctx, test.salt, test.salt_size), 1);
+  EXPECT_EQ(EVP_PKEY_CTX_add1_hkdf_info(pctx, test.info, test.info_size), 1);
```

Comment:
Why not use ASSERT_TRUE()? Also on l. 1647?

Reply:
done!